### PR TITLE
Refactor reflect_helpers a bit to support other types

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

**Comment by matthchr** [comment]
More prep work for secrets

File: v2/internal/reflecthelpers/reflect_helpers.go [hunk 1]
```diff
@@ -36,26 +36,41 @@ func DeepCopyInto(in client.Object, out client.Object) {
...
 			return nil
 		}
 
 		return IdentityVisitStruct(this, it, ctx)
 	}
 
-	err := visitor.Visit(transformer, nil)
+	err := visitor.Visit(obj, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "scanning for %s", t.String())
```
Comment (theunrepentantgeek) [inline v2/internal/reflecthelpers/reflect_helpers.go:58]:
```suggestion
		return nil, errors.Wrapf(err, "scanning for references of type %s", t.String())
```

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/reflecthelpers/reflect_helpers.go (hunk 1 above)
Comment (matthchr) [reply v2/internal/reflecthelpers/reflect_helpers.go:58]: Done

**Pull request merged**